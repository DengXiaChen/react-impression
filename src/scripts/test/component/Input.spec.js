import React from 'react';
import { shallow, mount } from 'enzyme';
import { Input } from '../../components/impression';

const setup = () => {
    const props = {
        type: 'text',
        placeholder: '请输入',
        defaultValue: 'default',
        clearable: true,
        pill: true,
    };

    const wrapper = shallow(<Input {...props} />);
    const wrapper1 = mount(<Input {...props} />);

    return {
        props,
        wrapper,
        wrapper1,
    };
};

describe('Input', () => {
    const { wrapper, wrapper1 } = setup();

    it('render', () => {
        // expect(wrapper1.find('i').exists()).toBe(true);
        expect(wrapper1.find('input').prop('placeholder')).toBe('请输入');
        expect(wrapper1.find('input').props().defaultValue).toBe('default');
        expect(wrapper1.find('input').node.value).toBe('default');
        expect(wrapper1.find('.input-pill').exists()).toBe(true);
    });

    it('input', () => {
        const input = wrapper1.find('input');

        input.simulate('focus');
        input.simulate('keyDown', {
            which: 83,
        });
        expect(wrapper1.find('input').props().defaultValue).toBe('default');
        expect(wrapper1.find('input').node.value).toBe('default');
    });

    it('ui', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
