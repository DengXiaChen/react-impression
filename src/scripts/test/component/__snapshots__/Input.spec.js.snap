// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="input"
>
    <input
        className="form-control input-pill input-field"
        defaultValue="default"
        disabled={false}
        placeholder="请输入"
        style={undefined}
        type="text"
        value={undefined}
    />
</div>,
  "nodes": Array [
    <div
      className="input"
>
      <input
            className="form-control input-pill input-field"
            defaultValue="default"
            disabled={false}
            placeholder="请输入"
            style={undefined}
            type="text"
            value={undefined}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Input
        clearable={true}
        defaultValue="default"
        disabled={false}
        pill={true}
        placeholder="请输入"
        type="text"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Input {
        "_reactInternalInstance": [Circular],
        "clearInputHandle": [Function],
        "context": Object {},
        "hideClearHandle": [Function],
        "hideOptionsHandle": [Function],
        "props": Object {
          "clearable": true,
          "defaultValue": "default",
          "disabled": false,
          "pill": true,
          "placeholder": "请输入",
          "type": "text",
        },
        "refs": Object {},
        "selectOptionsHandle": [Function],
        "showClearHandle": [Function],
        "showOptionHandle": [Function],
        "state": Object {
          "showClear": false,
          "showOption": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="input"
>
          <input
                    className="form-control input-pill input-field"
                    defaultValue="default"
                    disabled={false}
                    placeholder="请输入"
                    style={undefined}
                    type="text"
                    value={undefined}
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="input"
>
          <input
                    className="form-control input-pill input-field"
                    defaultValue="default"
                    disabled={false}
                    placeholder="请输入"
                    style={undefined}
                    type="text"
                    value={undefined}
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Input
    clearable={true}
    defaultValue="default"
    disabled={false}
    pill={true}
    placeholder="请输入"
    type="text"
/>,
}
`;
